joblib/test/test_parallel.py

```diff
@@ -1260,13 +1260,23 @@ def test_memmapping_leaks(backend, tmpdir):
 
     # Make sure that the shared memory is cleaned at the end when we exit
     # the context
-    assert not os.listdir(tmpdir)
+    for _ in range(100):
```

Comment:
This test failed for no apparent reason at the last commit. I am tempted to re-write it as an isolated subprocess script to limit flakiness and better debug it. WDYT @ogrisel ?